### ISSUE TITLE
Abilty to output results for each Describe Block to screen and seperate XML files

### ISSUE THREAD

**Issue by kilasuit** [pester/Pester#445]
This may just be because of my relative newness to using Pester but I would find it beneficial to be able to output to the screen a report of the passed/failed/skipped/pending tests for each Describe Block before starting the next Describe block whilst also being able to output this to an XML file for future reference.

The reason behind this is that I have a decent sized Script Library and I am intending to work through it in chunks to build up a better standard test but I am wanting to find out which of the chunks is going to be the biggest headache so that i can get them over and done with sooner rather than later.

I would be encapsulating the Describe block within something like the following
```powershell
$scripts = Get-ChildItem Scripts:\Archived -recurse
    If ($scripts.Where({$_.LastWriteTime -gt ((Get-Date).AddDays(-10))}).count -gt 1) {
        Describe "Tests for Archived" { 
            "Tests to go here" }
}
```

Is this currently feasible with just 1 Pester Test script or will I need to split this into separate Test Scripts which although is possible it would mean having a few extra copies of the same code which I'm looking to reduce as much as possible.

**Comment by dlwyatt** [comment]
The console output doesn't do this at the moment, but the information you need is in the XML file or the -PassThru object.  For example, if you just want to know the number of non-successful tests per describe, you could do something like this:

```posh
$results = Invoke-Pester -PassThru

$describes = $results.TestResult | Group-Object -Property Describe
$describeSummary = $describes | Select-Object -Property Name, @{ Name = 'FailCount'; Expression = { $_.Group | Where-Object { -not $_.Passed } | Measure-Object | Select-Object -ExpandProperty Count } }
$describeSummary | Sort-Object -Property FailCount -Descending
```

**Comment by kilasuit** [comment]
Thanks for this @dlwyatt  I missed the PassThru Parameter completely and the example you've provided works like a charm for what I want it to do :-)
